include/openssl/bio.h

```diff
@@ -724,7 +724,7 @@ void BIO_copy_next_retry(BIO *b);
  * long BIO_ghbn_ctrl(int cmd,int iarg,char *parg);
  */
 
-# ifdef __GNUC__
+# if defined(__GNUC__) && __STDC_VERSION__ >= 199901L
```

Comment:
There should even be `defined(__STDC_VERSION__)`, because --strict-warnings check even for undefined macros.